### ISSUE TITLE
Import audio

### ISSUE THREAD

**Issue by BoatsAreRockable** [BoatsAreRockable/Boats-Animator#161]
The ability to import audio files should eventually exist to help with matching animations to dialogue etc. 

An audio speed option should be available in conjunction that is independent of the FPS change feature (ie even though FPS = 15, if audio speed = 50%, playback at effectively 7.5 FPS).

**Comment by rioforce** [comment]
Was just searching around the web and came across this. I thought I'd link it here.

https://wavesurfer-js.org/

Seems pretty nifty to add some time down the road when we want to add waveforms underneath our frame icons, or on an x-sheet (depending on which way we go with development).

**Comment by charlielee** [comment]
Should be noted that I recieved quite a few user requests for this feature recently

**Comment by AndreasDemenagas** [comment]
> Should be noted that I have received quite a few user requests for this feature recently

Yeah, I saw the discussion on BiM. I still need to do some research on how JS handles audio & and possible ways to implement it in the app.

**Comment by AndreasDemenagas** [comment]
By playing around with [https://wavesurfer-js.org/](https://wavesurfer-js.org/ ), as suggested by rioforce above, I was able to get the following result with less than 10 lines of JS + HTML. They provide a really powerful framework. 

We should set the `skipLength` property of the waveform to something proportional to their frame rate (1/15th of a second if animating on 15 FPS etc..), and then call the `skipForward()` and `skipBackward()` functions when new frames are being added or when the user is moving through frames in the timeline. 

![Screenshot 2020-05-08 at 13 38 34](https://user-images.githubusercontent.com/55033328/81398126-4b3dfc80-9131-11ea-8118-dd3a8f509cef.png)

**Comment by AndreasDemenagas** [comment]
Began trying to implement this in the `import-audio` branch. No promises still :)

**Comment by charlielee** [comment]
https://css-tricks.com/introduction-web-audio-api/

**Comment by charlielee** [comment]
Some of the data structure changes to support this are:

* **Take** - add an `audioTracks: Track[]` field
* **FileRefType** - add `AUDIO` and `BLANK` file types. A blank file enables audio represents space in the track before/after the audio file

the length (in frames) of an audio TrackItem will need to be calculated on import and recalculated if the take's frame rate is changed.

**Comment by charlielee** [comment]
A few other notes:

* Should be able to move the start position of imported audio
* Audio should be shown on the X-sheet.
* Perhaps only allow one audio track to start for simplicity?
* May later wish to make audio trimmable
